docs/spec/callables.rst

```diff
@@ -347,60 +403,394 @@ generate errors in such cases.
 Callable
 --------
 
-Frameworks expecting callback functions of specific signatures might be
-type hinted using ``Callable[[Arg1Type, Arg2Type], ReturnType]``.
-Examples::
+The ``Callable`` special form can be used to specify the signature of
+a function within a type expression. The syntax is
+``Callable[[Param1Type, Param2Type], ReturnType]``. For example::
+
+    from collections.abc import Callable
+
+    def func(cb: Callable[[int], str]) -> None:
+        ...
+
+    x: Callable[[], str]
+
+Parameters specified using ``Callable`` are assumed to be positional-only.
+The ``Callable`` form provides no way to specify keyword-only parameters,
+variadic parameters, or default argument values. For these use cases, see
+the section on `Callback protocols`_.
+
+Meaning of ``...`` in ``Callable``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``Callable`` special form supports the use of ``...`` in place of the
+list of parameter types. This indicates that the type is consistent with
+any input signature. Just as ``Any`` means "any conceivable type that could be
+compatible", ``(...)`` means "any conceivable set of parameters that could be
+compatible"::
+
+    cb1: Callable[..., str]
+    cb1 = lambda x: str(x)  # OK
+    cb1 = lambda : ""  # OK
+
+    cb2: Callable[[], str] = cb1  # OK
+
+A ``...`` can also be used with ``Concatenate``. In this case, the parameters
+prior to the ``...`` are required to be present in the input signature and
+be compatible in kind and type, but any additional parameters are permitted::
+
+    cb3: Callable[Concatenate[int, ...], str]
+    cb3 = lambda x: str(x)  # OK
+    cb3 = lambda a, b, c: str(a)  # OK
+    cb3 = lambda : ""  # Error
+    cb3 = lambda *, a: str(a)  # Error
+
+
+If the input signature in a def statement includes both a ``*args`` and
+``**kwargs`` parameter and both are typed as ``Any`` (explicitly or implicitly
+because it has no annotation), a type checker should treat this as the
+equivalent of ``...``. Any other parameters in the signature are unaffected
+and are retained as part of the signature::
+
+    def func1(*args: Any, **kwargs: Any) -> None:
+        pass
+
+    def func2(a: int, /, *args, **kwargs) -> None:
+        pass
 
-  from collections.abc import Callable
+    def func3(a: int, *args: Any, **kwargs: Any) -> None:
+        pass
 
-  def feeder(get_next_item: Callable[[], str]) -> None:
-      # Body
+    class Proto1[**P](Protocol):
+        def __call__(self, a: int, *args: P.args, **kwargs: P.kwargs) -> None: ...
 
-  def async_query(on_success: Callable[[int], None],
-                  on_error: Callable[[int, Exception], None]) -> None:
-      # Body
+    assert_type(func1, Callable[..., None])  # OK
```

Comment:
I think a reasonable type checker could reject this call on the grounds that it knows `func1` is an instance of the concrete function type, rather than a generic callable. Function objects have several attributes that callables in general don't.

Safer (but clunkier) ways to demonstrate the behavior could include the override example below, and defining a callback protocol that takes `*args: Any, **kwargs: Any`.